torch_ttnn/passes/memory_pass.py

```diff
@@ -0,0 +1,240 @@
+                    last_assigned_tid += 1
+            # For output
+            if node not in mm.node_to_tid_map:
+                mm.node_to_tid_map[node] = last_assigned_tid
+                last_assigned_tid += 1
+
+    # print(f"Tensor IDs for nodes on device:")
+    # for key, value in mm.node_to_tid_map.items():
+    #     print(f"Node: {key} - {value}")
+
+    # Tracks the total compute memory of the model
+    mm.logs += f"Total tt compute nodes: {tt_compute_count}\n"
+    print(f"Total tt compute nodes: {tt_compute_count}")
+    i = 0
+    overflow_ops = []
+
+    for node in nodes:
+        on_device = False
+        # Has to be a ttnn op
+        if is_tt_compute(node):
+            mm.logs += "---------------------------------------------------\n"
+            print("\n---------------------------------------------------")
+            mm.logs += f"tt compute node numbered: {i}\n"
+            print(f"tt compute node numbered: {i}")
+            i += 1
+            # Exception: Full node has no input
+            # Assumption: Full node outputs a tensor on device
+            if node.target == ttnn.full:
+                on_device = True
+            total_input_size = 0
+            # Input & Output tensor IDs for ttnn op
+            ttnn_ops_tids = []
+            for input_node in node.all_input_nodes:
+                # If input tensor on device or coming as output from another ttnn on device op
+                if op_registry.is_input_tensor_on_device(input_node):
+                    on_device = True
+                    input_shape = op_registry.get_shape(input_node)
+                    input_dtype = op_registry.get_dtype(input_node)
+                    input_size = op_registry.get_size(input_shape, input_dtype)
+                    total_input_size += input_size
+
+                    assert (
+                        input_node in mm.node_to_tid_map
+                    ), "Tensor ID not allocated for one of the inputs!"
+
+                    tid = mm.node_to_tid_map[input_node]
+                    ttnn_ops_tids.append(tid)
+                    mm.tensor_size_of[tid] = input_size
+
+                    start_addr = last_assigned_addr
+                    last_assigned_addr += input_size
+                    end_addr = last_assigned_addr
+                    mm.tid_to_addr_map_in_sram[tid] = (start_addr, end_addr)
+
+                    if tid not in mm.tensors_on_device:
+                        mm.used_sram += input_size
+                        mm.free_sram -= input_size
+                        mm.tensors_on_device.append(tid)
+
+                    if (tid, input_size) not in on_device_meta:
+                        on_device_meta.append((tid, input_size))
+
+            if on_device:
+                output_shape = op_registry.get_shape(node)
+                output_dtype = op_registry.get_dtype(node)
+                output_size = op_registry.get_size(output_shape, output_dtype)
```

Comment:
```
output_size = op_registry.get_size(output_shape, output_dtype)
```

Why is this a function of `op_registry`?

Reply:
Removed from op_registry class, now directly accessible in mem_utils module.